CMakeLists.txt

```diff
@@ -54,6 +72,12 @@ link_directories(
   ${OFED_PREFIX}/lib
   ${LEVELDB_PREFIX}/lib
 )
+if(${FREEBSD})
```

Comment:
i did read this also, the reason why i'd recommend `if(foo)` is for the sake of simplicity and consistency. they are the equivalent under most circumstances as explained by the document you quote.